### PR TITLE
feat(core): Manually run consents in gen scripts

### PR DESCRIPTION
**Pull request by mattjoke**
* Fixed getHashedHierarchicalData to work with new parameter
* Fixed getHashedDataWithGroups to work with new parameter
* Added openpai optional parameter for the gen scripts

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java [hunk 1]
```diff
@@ -106,12 +110,14 @@ private GenResourceDataNode getDataForResource(Resource resource) {
 		} else {
 			members = sess.getPerunBl().getResourcesManagerBl().getAllowedMembers(sess, resource);
 		}
-		if (BeansUtils.getCoreConfig().getForceConsents()) {
-			// remove the members without granted consents on required attributes
-			members.removeIf(member -> !membersWithConsent.contains(member));
-		} else {
-			// we skipped this part if consents were required, so add them now
-			membersWithConsent.addAll(members);
+		if (consentEval) {
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java:113]: Join into one IF statement. We still want to run the code in this else block if consentEval is false.

File: perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/GroupsHashedDataGenerator.java [hunk 2]
```diff
@@ -87,7 +89,9 @@ public HashedGenData generateData() {
 			} else {
 				membersToEvaluate = sess.getPerunBl().getFacilitiesManagerBl().getAllowedMembers(sess, facility, service);
 			}
-			membersWithConsent.addAll(sess.getPerunBl().getConsentsManagerBl().evaluateConsents(sess, service, facility, membersToEvaluate));
+			if (consentEval) {
```
Comment (xflord) [inline perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/GroupsHashedDataGenerator.java:92]: Join into one if statement on line 85. We want to check if we force consents globally AND for that specific service

**Review by xflord — CHANGES_REQUESTED** [review]
Also update the comments/documentation in ServicesAgent.pm to include the new parameter. Some of the tests are failing, because you forgot to update them with the new consentEval parameter -  try to run relevant tests yourself before pushing.

File: perun-cli/Perun/Agent.pm [hunk 3]
```diff
@@ -56,6 +57,12 @@ use constant {
 	WRONG_AGENT_VERSION   => "Version of the command tools and Perun server mismatch",
 };
 
+$ENV{"CONSENT_EVAL"} = 0;
+GetOptions("consent-eval|c" => sub{
+	$ENV{"CONSENT_EVAL"} = 1;
+});
```
[the remark this answers is not in the document]
Comment (mattjoke) [reply perun-cli/Perun/Agent.pm:63]: You are right, this approach enables the consent eval in every script using Agent.pm.

**Review by xflord — CHANGES_REQUESTED** [review]
Also as @Johaney-s probably already told you, we need to make sure that automatic calls of services set consentEval to true (probably?). Ask around how/where this is done.

File: perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java [hunk 4]
```diff
@@ -72,7 +73,7 @@ public HashedGenData generateData() {
 		List<Resource> resources =
 				sess.getPerunBl().getFacilitiesManagerBl().getAssignedResources(sess, facility, null, service);
 
-		if (BeansUtils.getCoreConfig().getForceConsents()) {
+		if (consentEval && BeansUtils.getCoreConfig().getForceConsents()) {
```
Comment (Johaney-s) [inline perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java:76]: we\ll want to filter out members that do not have consent even if calling it without the consentEval flag, we just want to skip creation of new consents in that case

File: perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java [hunk 5]
```diff
@@ -106,7 +107,7 @@ private GenResourceDataNode getDataForResource(Resource resource) {
 		} else {
 			members = sess.getPerunBl().getResourcesManagerBl().getAllowedMembers(sess, resource);
 		}
-		if (BeansUtils.getCoreConfig().getForceConsents()) {
+		if (consentEval && BeansUtils.getCoreConfig().getForceConsents()) {
```
Comment (Johaney-s) [inline perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java:110]: we'll want to filter out members that do not have consent even if calling it without the consentEval flag, we just want to skip creation of new consents in that case

**Review by Johaney-s — APPROVED** [review]

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java [hunk 6]
```diff
@@ -343,7 +351,7 @@ private boolean hasValidConsent(PerunSession sess, Member member, ConsentHub con
 			return true;
 		} else if (revokedConsent != null && revokedConsent.getAttributes().containsAll(requiredAttributes)) {
 			return false;
-		} else if (unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes)) {
+		} else if (unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes) && consentEval) {
```
Comment (balcirakpeter) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java:354]: Please add extra parentheses so the condition is more understandable.
```suggestion
		} else if ((unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes)) && consentEval) {
```

**Review by balcirakpeter — CHANGES_REQUESTED** [review]
Please update also Javadoc in `ServicesAgent.pm`  for `getHashedData` subroutines.

**Pull request merged**